### PR TITLE
cam-254-oauth_appexchange: allow apps to exchange an existing app token

### PR DESCRIPTION
**Pull request by mikew-personal**
#### Ticket
[CAM-254](https://vimean.atlassian.net/browse/CAM-254)

#### Ticket Summary
The API team has created a new endpoint that allows an application to exchange an access toke that was issued for a different Vimeo app for a new access token that is issued to the calling application.  This will allow apps like Cameo and DayTripper to automatically log a user in using a token that has already been saved by the Vimeo app.  This endpoint is also a crucial component of the Cameo -> Vimeo networking switchover work.

#### Implementation Summary
I added a new method to Request+Authentication that creates the proper request for the oauth/appexchange endpoint.  A new method (appTokenExchange(accessToken:completion:)) was added to the AuthenticationController class for client applications to use when exchanging a token. 

#### How to Test
1) Log in with client credentials
2) call AuthenticationController.appTokenExchange and pass in a valid Vimeo access token.  The completion handler should return a valid VIMAccount containing a new access token for the calling application to use.

### CODE REVIEW

**Comment by mikew-personal** [comment]
@huebnerob there's currently a bug in the API endpoint that fails to exchange Cameo tokens that were recently created on the Cameo server.  Brian has a fix for this issue that will get pushed to prod some time this afternoon.  Testing the endpoint against tokens issued from the Vimeo app should work fine.

**Comment by mikew-personal** [comment]
@huebnerob,  Brian pushed his fix to prod.  The endpoint is now working correctly for all Cameo tokens.

File: VimeoNetworking/VimeoNetworking/AuthenticationController.swift [hunk 1]
```diff
@@ -211,6 +211,13 @@ final public class AuthenticationController
         self.authenticate(request: request, completion: completion)
     }
     
+    public func appTokenExchange(accessToken accessToken: String, completion: AuthenticationCompletion)
```
Comment (huebnerob) [inline VimeoNetworking/VimeoNetworking/AuthenticationController.swift]: Can we document this method? Pull in recent changes from `master` to see the format

File: VimeoNetworking/VimeoNetworking/Request+Authentication.swift [hunk 2]
```diff
@@ -103,6 +105,13 @@ public extension Request where ModelType: VIMAccount
         
         return Request(method: .POST, path: AuthenticationPathPinCodeAuthorize, parameters: parameters, cacheFetchPolicy: .NetworkOnly, shouldCacheResponse: false)
     }
+    
+    public static func appTokenExchangeRequest(accessToken accessToken: String) -> Request
```
Comment (huebnerob) [inline VimeoNetworking/VimeoNetworking/Request+Authentication.swift]: same here, can we document this?

**Comment by huebnerob** [comment]
@mikew-personal Looks great!

I recently (today) merged documentation for all of `VimeoNetworking` so I'd love it if these new methods were documented too.  When you merge with master, it'll all be there

**Pull request merged**